### PR TITLE
Update verify-openapi-spec script to check for extra generated spec

### PR DESCRIPTION
**Pull request by mbohlool**
hack/verify-openapi-spec.sh only check for existing spec changes. If for some reason (here most probably I forgot to delete a file in api/openapi-spec folder in #35388 after a rebase) there is an old spec exists in the spec folder, it won't panic but it should. This resulted in an unused out of date v1.spec file in the api/openapi-spec folder that this PR also removes.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/35841)

### CODE REVIEW

File: hack/verify-openapi-spec.sh [hunk 1]
```diff
@@ -32,14 +32,15 @@ TMP_SPECROOT="${KUBE_ROOT}/_tmp/openapi-spec"
 _tmp="${KUBE_ROOT}/_tmp"
 
 mkdir -p "${_tmp}"
-trap "rm -rf ${_tmp}" EXIT SIGINT
 cp -a "${SPECROOT}" "${TMP_SPECROOT}"
+trap "cp -a ${TMP_SPECROOT} \"${KUBE_ROOT}/api\"; rm -rf ${_tmp}" EXIT SIGINT
```
Comment (caesarxuchao) [inline hack/verify-openapi-spec.sh]: Will `cp -a ${TMP_SPECROOT}/* ${SPECROOT}` or `cp -a ${TMP_SPECROOT} ${SPECROOT}/..` work?

**Comment by caesarxuchao** [comment]
One comment and adding back the -N, then lgtm. Feel free to self apply the label.

File: hack/verify-openapi-spec.sh (hunk 1 above)
Comment (mbohlool) [inline hack/verify-openapi-spec.sh]: Yes, it is works and better. I just (blindly) copy/paster the command on (old) line 42. Would use your first suggestion.

**Comment by mbohlool** [comment]
@k8s-bot node e2e test this issue #ignore

**Comment by mbohlool** [comment]
@k8s-bot node e2e test this issue #35004

**Comment by nikhiljindal** [comment]
@mbohlool Dont you need to make the same change in verify-federation-openapi-spec?

**Comment by nikhiljindal** [comment]
ok I see, we dont have verify-federation-openapi-spec, only update-federation-openapi-spec :)
We should add it!

**Comment by mbohlool** [comment]
yep, we should add verify script but let me add it in another PR. This PR get rebased often because people edit v1.json spec file.

**Comment by nikhiljindal** [comment]
oh yes, dont need to block this PR :)

**Pull request merged**